test/manual/mongodb_oidc.prose.test.ts

```diff
@@ -452,5 +452,153 @@ describe('MONGODB-OIDC', function () {
         });
       });
     });
+
+    // The driver MUST test reauthentication with MONGODB-OIDC for a read operation.
+    describe('6. Reauthentication', function () {
+      let refreshInvokations = 0;
+      let findStarted = 0;
+      let findSucceeded = 0;
+      let findFailed = 0;
+      let saslStarted = 0;
+      let saslSucceeded = 0;
+      let client;
+      let collection;
+      const cache = OIDC_WORKFLOWS.get('callback').cache;
+
+      // - Create request and refresh callbacks that return valid credentials that
+      //   will not expire soon.
+      const requestCallback = async () => {
+        const token = await readFile(`${process.env.OIDC_TOKEN_DIR}/test_user1`, {
+          encoding: 'utf8'
+        });
+        return { accessToken: token, expiresInSeconds: 300 };
+      };
+
+      const refreshCallback = async () => {
+        const token = await readFile(`${process.env.OIDC_TOKEN_DIR}/test_user1`, {
+          encoding: 'utf8'
+        });
+        refreshInvokations++;
+        return { accessToken: token, expiresInSeconds: 300 };
+      };
+
+      const commandStarted = event => {
+        if (event.commandName === 'find') {
+          findStarted++;
+        }
+        if (event.commandName === 'saslStart') {
+          saslStarted++;
+        }
+      };
+
+      const commandSucceeded = event => {
+        if (event.commandName === 'find') {
+          findSucceeded++;
+        }
+        if (event.commandName === 'saslStart') {
+          saslSucceeded++;
+        }
+      };
+
+      const commandFailed = event => {
+        if (event.commandName === 'find') {
+          findFailed++;
+        }
+      };
+
+      before(function () {
+        // - Clear the cache
+        cache.clear();
+        // - Create a client with the callbacks and an event listener capable of
+        //   listening for SASL commands
+        //
+        // - TODO(NODE-3494): Driver does not observe sensitive commands.
+        client = new MongoClient('mongodb://test_user1@localhost/?authMechanism=MONGODB-OIDC', {
+          authMechanismProperties: {
+            REQUEST_TOKEN_CALLBACK: requestCallback,
+            REFRESH_TOKEN_CALLBACK: refreshCallback
+          },
+          monitorCommands: true
+        });
+        client.on('commandStarted', commandStarted);
+        client.on('commandSucceeded', commandSucceeded);
+        client.on('commandFailed', commandFailed);
+        collection = client.db('test').collection('test');
```

Comment:
General question about these test. Do you know why they are simply asserting on the correct number of events?  It seems like the ordering of events should be exact:

find start, find failed, sasl start, sasl suceeded, find start, find succeeded

asserting on the counts of events does nothing to assert on the correct behavior - this ordering would also pass (and is the ordering that would happen in a normal retry, not reauth)

sasl start, sasl succeeded, find start, find failed, find start, find succeeded

Reply:
DRIVERS-2415 has a lot of changes to the prose tests around this area. May be better to wait for that to finish and then re-evaluate.

Reply:
sounds good - I left a comment on the PR.

Reply:
can you add a ticket to the OIDC support epic to incorporate changes from that PR (if we don't have one)?  I don't want to lose track of that work.